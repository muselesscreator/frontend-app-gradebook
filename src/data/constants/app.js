import { StrictDict } from 'utils';

export const modalFieldKeys = StrictDict({
  adjustedGradePossible: 'adjustedGradePossible',
  adjustedGradeValue: 'adjustedGradeValue',
  assignmentName: 'assignmentName',
  reasonForChange: 'reasonForChange',
  todaysDate: 'todaysDate',
  updateModuleId: 'updateModuleId',
  updateUserId: 'updateUserId',
  updateUserName: 'updateUserName',
  open: 'open',
});

export const localFilterKeys = StrictDict({
  assignmentGradeMax: 'assignmentGradeMax',
  assignmentGradeMin: 'assignmentGradeMin',
  courseGradeMax: 'courseGradeMax',
  courseGradeMin: 'courseGradeMin',
});
