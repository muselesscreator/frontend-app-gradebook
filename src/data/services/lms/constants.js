import { StrictDict } from 'utils';

export const pageSize = 25;
export const historyRecordLimit = 5;

export const paramKeys = StrictDict({
  cohortId: 'cohort_id',
  pageSize: 'page_size',
  userContains: 'user_contains',
  enrollmentMode: 'enrollment_mode',
  assignment: 'assignment',
  assignmentGradeMin: 'assignmentGradeMin',
  assignmentGradeMax: 'assignmentGradeMax',
  courseGradeMin: 'courseGradeMin',
  courseGradeMax: 'courseGradeMax',
  excludedCourseRoles: 'excluded_course_roles',
});
