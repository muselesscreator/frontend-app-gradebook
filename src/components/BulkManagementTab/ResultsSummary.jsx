/* eslint-disable react/sort-comp, react/button-has-type */
import React from 'react';
import PropTypes from 'prop-types';

import { Hyperlink, Icon } from '@edx/paragon';
import { Download } from '@edx/paragon/icons';

import { bulkGradesUrlByCourseAndRow } from 'data/constants/api';

const ResultsSummary = ({
  courseId,
  rowId,
  text,
}) => (
  <Hyperlink
    href={bulkGradesUrlByCourseAndRow(courseId, rowId)}
    destination="www.edx.org"
    target="_blank"
    rel="noopener noreferrer"
  >
    <Icon icon={Download} />
    {text}
  </Hyperlink>
);

ResultsSummary.propTypes = {
  courseId: PropTypes.string.isRequired,
  rowId: PropTypes.number.isRequired,
  text: PropTypes.string.isRequired,
};

export default ResultsSummary;
