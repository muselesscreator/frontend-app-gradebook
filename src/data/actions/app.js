import { StrictDict } from 'utils';
import { modalFieldKeys, localFilterKeys } from 'data/constants/app';
import { createActionFactory } from './utils';

export const dataKey = 'app';
const createAction = createActionFactory(dataKey);

const closeModal = createAction('closeModal');
/**
 * setCourseId(courseId)
 * loads courseID into local app state
 * @param {string} courseId - course ID from app context
 */
const setCourseId = createAction('setCourseId');
/**
  * setModalStateFromTable({ userEntry, subsection })
  * sets modalState
  * */
const setModalStateFromTable = createAction('setModalStateFromTable');
/**
 * setSearchValue(searchValue)
 * sets searchValue in local state
 * @param {string} searchValue - search input text
 */
const setSearchValue = createAction('setSearchValue');

const filterReducer = (payload) => (obj, key) => (
  (payload[key] !== undefined) ? { ...obj, [key]: payload[key] } : obj
);

/**
 * setLocalFilter(filterObject)
 * sets a number of localFilterKey fields on localFilters
 * @param {object} filterObject - an object of filter values
 */
const setLocalFilter = createAction('setLocalFilter', (filterObject) => ({
  payload: Object.keys(localFilterKeys).reduce(filterReducer(filterObject), {}),
}));

/**
 * setModalState(modalState)
 * sets a number of modalFieldKey fields on modalState
 * @param {object} modalState - an object of modal state values
 */
const setModalState = createAction('setModalState', (modalState) => ({
  payload: Object.keys(modalFieldKeys).reduce(filterReducer(modalState), {}),
}));

export default StrictDict({
  closeModal,
  setCourseId,
  setModalState,
  setSearchValue,
  setLocalFilter,
  setModalStateFromTable,
});
