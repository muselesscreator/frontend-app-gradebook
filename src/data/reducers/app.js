import initialFilters from '../constants/filters';
import { formatDateForDisplay } from '../actions/utils';
import actions from '../actions/app';
import filterActions from '../actions/filters';

const initialState = {
  courseId: '',
  filters: {
    assignmentGradeMax: initialFilters.assignmentGradeMax,
    assignmentGradeMin: initialFilters.assignmentGradeMin,
    courseGradeMax: initialFilters.courseGradeMax,
    courseGradeMin: initialFilters.courseGradeMin,
  },
  modalState: {
    open: false,
    adjustedGradePossible: '',
    adjustedGradeValue: 0,
    assignmentName: '',
    reasonForChange: '',
    todaysDate: '',
    updateModuleId: null,
    updateUserId: null,
    updateUserName: null,
  },
  searchValue: '',
};

const app = (state = initialState, { type, payload }) => {
  switch (type) {
    case actions.closeModal.toString():
      return { ...state, modalState: { ...initialState.modalState } };
    case actions.setCourseId.toString():
      return { ...state, courseId: payload };
    case actions.setLocalFilter.toString():
      return {
        ...state,
        filters: { ...state.filters, ...payload },
      };
    case actions.setModalState.toString():
      return {
        ...state,
        modalState: { ...state.modalState, ...payload },
      };
    case actions.setModalStateFromTable.toString(): {
      const { subsection, userEntry } = payload;
      return {
        ...state,
        modalState: {
          adjustedGradeValue: '',
          open: true,
          reasonForChange: '',
          todaysDate: formatDateForDisplay(new Date()),

          adjustedGradePossible: subsection.attempted ? subsection.score_possible : '',
          assignmentName: `${subsection.subsection_name}`,
          updateModuleId: subsection.module_id,
          updateUserId: userEntry.user_id,
          updateUserName: userEntry.username,
        },
      };
    }
    case actions.setSearchValue.toString():
      return { ...state, searchValue: payload };
    // initialize the filter fields that are locally stored
    case filterActions.initialize.toString():
      return {
        ...state,
        filters: {
          assignmentGradeMax: payload.assignmentGradeMax,
          assignmentGradeMin: payload.assignmentGradeMin,
          courseGradeMax: payload.courseGradeMax,
          courseGradeMin: payload.courseGradeMin,
        },
      };
    // Reset only the filter fields that are stored locally
    case filterActions.reset.toString(): {
      return payload.filter(
        (filterName) => initialState.filters[filterName] !== undefined,
      ).reduce((obj, filterName) => ({
        ...obj,
        filters: {
          ...obj.filters,
          [filterName]: initialFilters[filterName],
        },
      }), { ...state });
    }
    default:
      return state;
  }
};

export { initialState };
export default app;
