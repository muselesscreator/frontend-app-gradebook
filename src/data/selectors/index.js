/* eslint-disable import/no-named-as-default-member, import/no-self-import */
import { StrictDict } from 'utils';
import LmsApiService from 'data/services/LmsApiService';

import * as module from '.';
import app from './app';
import assignmentTypes from './assignmentTypes';
import cohorts from './cohorts';
import filters from './filters';
import grades from './grades';
import roles from './roles';
import special from './special';
import tracks from './tracks';

/**
 * lmsApiServiceArgs(state)
 * Returns common lms api service request args.
 * @param {object} state - redux state
 * @return {object} lms api query params object
 */
export const lmsApiServiceArgs = (state) => ({
  cohort: cohorts.getCohortNameById(state, filters.cohort(state)),
  assignment: filters.selectedAssignmentId(state),
  assignmentType: filters.assignmentType(state),
  assignmentGradeMin: grades.formatMinAssignmentGrade(
    filters.assignmentGradeMin(state),
    { assignmentId: filters.selectedAssignmentId(state) },
  ),
  assignmentGradeMax: grades.formatMaxAssignmentGrade(
    filters.assignmentGradeMax(state),
    { assignmentId: filters.selectedAssignmentId(state) },
  ),
  courseGradeMin: grades.formatMinCourseGrade(filters.courseGradeMin(state)),
  courseGradeMax: grades.formatMaxCourseGrade(filters.courseGradeMax(state)),
});

/**
 * gradeExportUrl(state, options)
 * Returns the output of getGradeExportCsvUrl, applying the current includeCourseRoleMembers
 * filter.
 * @param {object} state - redux state
 * @param {object} options - options object of the form ({ courseId })
 * @return {string} - generated grade export url
 */
export const gradeExportUrl = (state, { courseId }) => (
  LmsApiService.getGradeExportCsvUrl(courseId, {
    ...module.lmsApiServiceArgs(state),
    excludeCourseRoles: filters.includeCourseRoleMembers(state) ? '' : 'all',
  })
);

/**
 * interventionExportUrl(state, options)
 * Returns the output of getInterventionExportUrl.
 * @param {object} state - redux state
 * @param {object} options - options object of the form ({ courseId })
 * @return {string} - generated intervention export url
 */
export const interventionExportUrl = (state, { courseId }) => (
  LmsApiService.getInterventionExportCsvUrl(
    courseId,
    module.lmsApiServiceArgs(state),
  )
);

/**
 * getHeadings(state)
 * Returns the table headings given the current assignmentType and assignmentLabel filters.
 * @param {object} state - redux state
 * @return {string[]} - array of table headings
 */
export const getHeadings = (state) => grades.headingMapper(
  filters.assignmentType(state) || 'All',
  filters.selectedAssignmentLabel(state) || 'All',
)(grades.getExampleSectionBreakdown(state));

/**
 * showBulkManagement(state, options)
 * Returns true iff the user has special access or bulk management is configured to be available
 * and the course has a masters track.
 * @param {object} state - redux state
 * @param {object} options - options object of the form ({ courseId })
 * @return {bool} - should show bulk management controls?
 */
export const showBulkManagement = (state, { courseId }) => (
  special.hasSpecialBulkManagementAccess(courseId)
  || (tracks.stateHasMastersTrack(state) && state.config.bulkManagementAvailable)
);

/**
 * shouldShowSpinner(state)
 * Returns true iff the user can view the gradebook and grades.showSpinner is true.
 * @param {object} state - redux state
 * @return {bool} - should show spinner?
 */
export const shouldShowSpinner = (state) => (
  roles.canUserViewGradebook(state)
  && grades.showSpinner(state)
);

/**
 * showBulkManagement(state, options)
 * Returns true iff the user has special access or bulk management is configured to be available
 * and the course has a masters track.
 * @param {object} state - redux state
 * @return {bool} - should show bulk management controls?
 */
export const showBulkManagement = (state) => (
  special.hasSpecialBulkManagementAccess(app.courseId(state))
  || (tracks.stateHasMastersTrack(state) && state.config.bulkManagementAvailable)
);

export default StrictDict({
  root: StrictDict({
    getHeadings,
    gradeExportUrl,
    interventionExportUrl,
    shouldShowSpinner,
  }),
  app,
  assignmentTypes,
  cohorts,
  filters,
  grades,
  roles,
  special,
  tracks,
});
