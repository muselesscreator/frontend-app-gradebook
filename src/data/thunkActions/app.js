import { StrictDict } from 'utils';

import actions from 'data/actions';
import { fetchGradeOverrideHistory } from './grades';
import { fetchRoles } from './roles';

export const setModalStateFromTable = ({ userEntry, subsection }) => (
  (dispatch) => {
    dispatch(fetchGradeOverrideHistory(subsection.module_id, userEntry.user_id));
    dispatch(actions.app.setModalStateFromTable({ subsection, userEntry }));
  }
);

export const initialize = (courseId, urlQuery) => (dispatch) => {
  dispatch(actions.app.setCourseId(courseId));
  dispatch(actions.filters.initialize(urlQuery));
  dispatch(fetchRoles());
};

export default StrictDict({
  initialize,
  setModalStateFromTable,
});
