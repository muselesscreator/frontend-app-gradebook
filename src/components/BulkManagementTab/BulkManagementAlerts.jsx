/* eslint-disable react/sort-comp, react/button-has-type */
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { Alert } from '@edx/paragon';

import * as appConstants from 'data/constants/app';
import selectors from 'data/selectors';

const { messages: { BulkManagementTab: messages } } = appConstants;

export const BulkManagementAlerts = ({ bulkImportError, uploadSuccess }) => (
  <>
    <Alert
      variant="danger"
      show={!!bulkImportError}
      dismissible={false}
    >
      {bulkImportError}
    </Alert>
    <Alert
      variant="success"
      show={uploadSuccess}
      dismissible={false}
    >
      {messages.successDialog}
    </Alert>
  </>
);

BulkManagementAlerts.defaultProps = {
  bulkImportError: '',
  uploadSuccess: false,
};

BulkManagementAlerts.propTypes = {
  // redux
  bulkImportError: PropTypes.string,
  uploadSuccess: PropTypes.bool,
};

export const mapStateToProps = (state) => ({
  bulkImportError: selectors.grades.bulkImportError(state),
  uploadSuccess: selectors.grades.uploadSuccess(state),
});

export default connect(mapStateToProps)(BulkManagementAlerts);
