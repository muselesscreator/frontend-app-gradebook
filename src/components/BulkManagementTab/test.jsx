/* eslint-disable import/no-named-as-default */
import React from 'react';
import { shallow } from 'enzyme';
import { Button, Form, Table } from '@edx/paragon';

import selectors from 'data/selectors';
import thunkActions from 'data/thunkActions';
import * as appConstants from 'data/constants/app';
import { BulkManagementTab, mapStateToProps, mapDispatchToProps } from '.';
import BulkManagementAlerts from './BulkManagementAlerts';
import ResultsSummary from './ResultsSummary';

const {
  bulkManagementColumns,
  messages: { BulkManagementTab: messages },
} = appConstants;
jest.mock('@edx/paragon', () => ({
  Button: () => 'Button',
  Form: {
    Control: () => 'Form.Control',
    Group: () => 'Form.Group',
    Label: () => 'Form.Label',
  },
  Table: () => 'Table',
}));
jest.mock('data/selectors', () => ({
  __esModule: true,
  default: {
    grades: {
      bulkImportError: jest.fn(state => ({ bulkImportError: state })),
      bulkManagementHistoryEntries: jest.fn(state => ({ historyEntries: state })),
    },
    root: {
      gradeExportUrl: jest.fn(state => ({ gradeExportUrl: state })),
    },
  },
}));
jest.mock('data/thunkActions', () => ({
  __esModule: true,
  default: {
    grades: { submitFileUploadFormData: jest.fn() },
  },

}));
jest.mock('./BulkManagementAlerts', () => 'BulkManagementAlerts');
jest.mock('./ResultsSummary', () => 'ResultsSummary');

describe('BulkManagementTab', () => {
  describe('component', () => {
    const entry1 = {
      originalFilename: 'blue.png',
      user: 'Eifel',
      timeUploaded: '65',
      resultsSummary: {
        rowId: 12,
        courseId: 'Da Bu Dee',
        text: 'Da ba daa',
      },
    };
    const entry2 = {
      originalFilename: 'allStar.jpg',
      user: 'Smashmouth',
      timeUploaded: '2000s?',
      resultsSummary: {
        courseId: 'rockstar',
        rowId: 2,
        text: 'all that glitters is gold',
      },
    };
    const props = {
      gradeExportUrl: 'fakeURL',
      bulkManagementHistory: [entry1, entry2],
    };
    let el;
    beforeEach(() => {
      props.submitFileUploadFormData = jest.fn();
    });
    describe('snapshot', () => {
      beforeEach(() => {
        el = shallow(<BulkManagementTab {...props} />);
      });
      const snapshotSegments = [
        'heading',
        'export form w/ alerts and file input',
        'import btn',
        'hints display',
        'formatted table',
      ];
      test(`snapshot - loads ${snapshotSegments.join(', ')}`, () => {
        el.instance().handleFileInputChange = jest.fn().mockName('this.handleFileInputChange');
        el.instance().fileInputRef = jest.fn().mockName('this.fileInputRef');
        el.instance().handleClickImportGrades = jest.fn().mockName('this.handleClickImportGrades');
        el.instance().formatHistoryRow = jest.fn(entry => entry.originalFilename);
        expect(el.instance().render()).toMatchSnapshot();
      });
      test('heading - h4 loaded from messages', () => {
        const heading = el.find('h4');
        expect(heading.text()).toEqual(messages.heading);
      });
      describe('alert form', () => {
        let form;
        beforeEach(() => {
          form = el.find('form');
        });
        test('post action points to gradeExportUrl', () => {
          expect(form.props().action).toEqual(props.gradeExportUrl);
          expect(form.props().method).toEqual('post');
        });
        test('loads BulkManagementAlerts', () => {
          expect(form.find(BulkManagementAlerts).length).toEqual(1);
        });
        describe('file input', () => {
          let fileInput;
          beforeEach(() => {
            fileInput = form.find(Form.Group);
          });
          test('group with controlId="csv"', () => {
            expect(fileInput.props().controlId).toEqual('csv');
          });
          test('label with csvUploadLabel message', () => {
            expect(fileInput.find(Form.Label).children().text()).toEqual(messages.csvUploadLabel);
          });
          test('file control with onChange from handleFileInputChange', () => {
            expect(
              fileInput.find(Form.Control).props().onChange,
            ).toEqual(el.instance().handleFileInputChange);
          });
        });
      });
      describe('import button', () => {
        let btn;
        beforeEach(() => {
          btn = el.find(Button);
        });
        test('handleClickImportGrade on click', () => {
          expect(btn.props().onClick).toEqual(el.instance().handleClickImportGrades);
        });
        test('text from messages.importBtn', () => {
          expect(btn.children().text()).toEqual(messages.importBtn);
        });
      });
      test('hints with break in between', () => {
        const hints = el.find('p');
        expect(hints.childAt(0).text()).toEqual(messages.hints[0]);
        expect(hints.childAt(1).is('br')).toEqual(true);
        expect(hints.childAt(2).text()).toEqual(messages.hints[1]);
      });
      describe('history table', () => {
        let table;
        beforeEach(() => {
          table = el.find(Table);
        });
        describe('data (from bulkManagementHistory.map(this.formatHistoryRow)', () => {
          const fieldAssertions = [
            'maps resultsSummay to ResultsSummary',
            'wraps filename and user',
            'forwards the rest',
          ];
          test(`snapshot: ${fieldAssertions.join(', ')}`, () => {
            expect(table.props().data).toMatchSnapshot();
          });
          test(fieldAssertions.join(', '), () => {
            const rows = table.props().data;
            expect(rows[0].resultsSummary).toEqual(<ResultsSummary {...entry1.resultsSummary} />);
            expect(rows[0].user).toEqual(<span className="wrap-text-in-cell">{entry1.user}</span>);
            expect(
              rows[0].filename,
            ).toEqual(<span className="wrap-text-in-cell">{entry1.originalFilename}</span>);
            expect(rows[1].resultsSummary).toEqual(<ResultsSummary {...entry2.resultsSummary} />);
            expect(rows[1].user).toEqual(<span className="wrap-text-in-cell">{entry2.user}</span>);
            expect(
              rows[1].filename,
            ).toEqual(<span className="wrap-text-in-cell">{entry2.originalFilename}</span>);
          });
        });
        test('columns from bulkManagementColumns', () => {
          expect(table.props().columns).toEqual(bulkManagementColumns);
        });
      });
    });
    describe('behavior', () => {
      let fileInput;
      beforeEach(() => {
        fileInput = jest.spyOn(el.instance(), 'fileInput', 'get');
      });
      describe('handleFileInputChange', () => {
        it('does nothing (does not fail) if fileInput has not loaded', () => {
          fileInput.mockReturnValue(null);
          el.instance().handleClickImportGrades();
        });
        it('calls fileInput.click if is loaded', () => {
          const click = jest.fn();
          fileInput.mockReturnValue({ click });
          el.instance().handleClickImportGrades();
          expect(click).toHaveBeenCalled();
        });
      });
      describe('handleClickImportGrades', () => {
        it('does nothing if file input has not loaded with files', () => {
          fileInput.mockReturnValue(null);
          el.instance().handleFileInputChange();
          expect(props.submitFileUploadFormData).not.toHaveBeenCalled();
          fileInput.mockReturnValue({ files: [] });
          el.instance().handleFileInputChange();
          expect(props.submitFileUploadFormData).not.toHaveBeenCalled();
        });
        it('calls submitFileUploadFormData and then clears fileInput if has files', () => {
          fileInput.mockReturnValue({ files: ['some', 'files'], value: 'a value' });
          const formData = { fake: 'form data' };
          jest.spyOn(el.instance(), 'formData', 'get').mockReturnValue(formData);
          const submit = jest.fn(() => ({ then: (thenCB) => { thenCB(); } }));
          el.setProps({
            submitFileUploadFormData: submit,
          });
          el.instance().handleFileInputChange();
          expect(submit).toHaveBeenCalledWith(formData);
          expect(el.instance().fileInput.value).toEqual(null);
        });
      });
    });
  });

  describe('mapStateToProps', () => {
    const testState = { a: 'simple', test: 'state' };
    let mapped;
    beforeEach(() => {
      mapped = mapStateToProps(testState);
    });
    test('bulkManagementHistory from grades.bulkManagementHistoryEntries', () => {
      expect(
        mapped.bulkManagementHistory,
      ).toEqual(selectors.grades.bulkManagementHistoryEntries(testState));
    });
    test('gradeExportUrl from root.gradeExportUrl', () => {
      expect(mapped.gradeExportUrl).toEqual(selectors.root.gradeExportUrl(testState));
    });
  });

  describe('mapDispatchToProps', () => {
    test('submitFileUploadFormData from thunkActions.grades', () => {
      expect(
        mapDispatchToProps.submitFileUploadFormData,
      ).toEqual(thunkActions.grades.submitFileUploadFormData);
    });
  });
});
