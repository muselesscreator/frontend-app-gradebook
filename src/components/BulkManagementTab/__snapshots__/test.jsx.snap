// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BulkManagementTab component snapshot history table data (from bulkManagementHistory.map(this.formatHistoryRow) snapshot: maps resultsSummay to ResultsSummary, wraps filename and user, forwards the rest 1`] = `
Array [
  Object {
    "filename": <span
      className="wrap-text-in-cell"
    >
      blue.png
    </span>,
    "resultsSummary": <ResultsSummary
      courseId="Da Bu Dee"
      rowId={12}
      text="Da ba daa"
    />,
    "timeUploaded": "65",
    "user": <span
      className="wrap-text-in-cell"
    >
      Eifel
    </span>,
  },
  Object {
    "filename": <span
      className="wrap-text-in-cell"
    >
      allStar.jpg
    </span>,
    "resultsSummary": <ResultsSummary
      courseId="rockstar"
      rowId={2}
      text="all that glitters is gold"
    />,
    "timeUploaded": "2000s?",
    "user": <span
      className="wrap-text-in-cell"
    >
      Smashmouth
    </span>,
  },
]
`;

exports[`BulkManagementTab component snapshot snapshot - loads heading, export form w/ alerts and file input, import btn, hints display, formatted table 1`] = `
<div>
  <h4>
    Use this feature by downloading a CSV for bulk management, overriding grades locally and coming back here to upload
  </h4>
  <form
    action="fakeURL"
    method="post"
  >
    <BulkManagementAlerts />
    <Group
      controlId="csv"
    >
      <Label>
        Upload Grade CSV
      </Label>
      <Control
        className="d-none"
        onChange={[MockFunction this.handleFileInputChange]}
        type="file"
      />
    </Group>
  </form>
  <Button
    onClick={[MockFunction this.handleClickImportGrades]}
    variant="primary"
  >
    Import Grades
  </Button>
  <p>
    Results appear in the table below
    <br />
    Grade processing may take a few seconds
  </p>
  <Table
    className="table-striped"
    columns={
      Array [
        Object {
          "columnSortable": false,
          "key": "filename",
          "label": "Gradebook",
          "width": "col-5",
        },
        Object {
          "columnSortable": false,
          "key": "resultsSummary",
          "label": "Download Summary",
          "width": "col",
        },
        Object {
          "columnSortable": false,
          "key": "user",
          "label": "Who",
          "width": "col-1",
        },
        Object {
          "columnSortable": false,
          "key": "timeUploaded",
          "label": "When",
          "width": "col",
        },
      ]
    }
    data={
      Array [
        "blue.png",
        "allStar.jpg",
      ]
    }
    hasFixedColumnWidths={true}
  />
</div>
`;
